code/modules/mob/living/basic/ruin_defender/cybersun_aicore.dm

```diff
@@ -0,0 +1,219 @@
+	/// Ability which fires da lightning bolt
+	var/datum/action/cooldown/mob_cooldown/lightning_strike
+	/// Ability which fires da big laser
+	var/datum/action/cooldown/mob_cooldown/targeted_mob_ability/donk_laser
+	//is this being used as part of the haunted trading post ruin? if true, will self destruct when boss dies
+	var/donk_ai_master = FALSE
+// list of stuff tagged to self destruct when this boss dies
+GLOBAL_LIST_EMPTY(selfdestructs_when_boss_dies)
+
+/mob/living/basic/cybersun_ai_core/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_IMMOBILIZED, INNATE_TRAIT)
+	var/static/list/death_loot = list(/obj/effect/temp_visual/cybersun_ai_core_death)
+	AddElement(/datum/element/death_drops, death_loot)
+	AddElement(/datum/element/relay_attackers)
+	var/static/list/innate_actions = list(
+		LIGHTNING_ABILITY_TYPEPATH = BB_CYBERSUN_CORE_LIGHTNING,
+		BARRAGE_ABILITY_TYPEPATH = BB_CYBERSUN_CORE_BARRAGE,
+	)
+	grant_actions_by_list(innate_actions)
+
+/mob/living/basic/cybersun_ai_core/death(gibbed)
+	if(donk_ai_master == TRUE)
+		//disable all the tripwire traps
+		for (var/obj/item/pressure_plate/puzzle/invisible_tripwire as anything in GLOB.selfdestructs_when_boss_dies)
+			qdel(invisible_tripwire)
+		//and the electric overload traps
+		for (var/obj/effect/overloader_trap as anything in GLOB.selfdestructs_when_boss_dies)
+			qdel(overloader_trap)
+		//then disable the AI defence holograms
+		for (var/obj/structure/holosign/barrier/cyborg/cybersun_ai_shield as anything in GLOB.selfdestructs_when_boss_dies)
+			qdel(cybersun_ai_shield)
+		//then the power generator
+		for (var/obj/machinery/power/smes/magical/cybersun as anything in GLOB.selfdestructs_when_boss_dies)
+			qdel(cybersun)
+	do_sparks(number = 5, source = src)
+	return ..()
+
+/obj/effect/temp_visual/cybersun_ai_core_death
+	icon = 'icons/mob/silicon/ai.dmi'
+	icon_state = "ai-red_dead"
+	duration = 2 SECONDS
+
+/obj/effect/temp_visual/cybersun_ai_core_death/Initialize(mapload)
+	. = ..()
+	playsound(src, 'sound/misc/metal_creak.ogg', vol = 100, vary = TRUE, pressure_affected = FALSE)
+	Shake(1, 0, 2 SECONDS)
+	addtimer(CALLBACK(src, PROC_REF(gib)), duration - 1, TIMER_DELETE_ME)
+
+/obj/effect/temp_visual/cybersun_ai_core_death/proc/gib()
+///dramatic death animations
+	playsound(loc, 'sound/effects/explosion2.ogg', vol = 75, vary = TRUE, pressure_affected = FALSE)
+	var/turf/my_turf = get_turf(src)
+	new /obj/effect/gibspawner/robot(my_turf)
+	for (var/mob/murderer in range(10, src))
+		if (!murderer.client || !isliving(murderer))
+			continue
+		playsound(loc, 'sound/effects/explosion2.ogg', vol = 75, vary = TRUE, pressure_affected = FALSE)
+		shake_camera(murderer, duration = 1.7 SECONDS, strength = 1)
+	for (var/mob/living in viewers(src, null))
+		if (!living.client || !isliving(living))
+			continue
+		playsound(loc, 'sound/effects/explosion2.ogg', vol = 55, vary = TRUE, pressure_affected = FALSE)
+		shake_camera(living, duration = 1 SECONDS, strength = 1)
```

Comment:
```suggestion
```
This is already being done above,

Reply:
The intent here is that the person who kills the boss gets screenshake even if they ducked behind a wall or fired the shot from out of visual range. Then everyone else gets screenshake only if they can see the ai core actually die. That way teams of people fighting the thing all get feedback. Is that not happening here? It's been a while since I touched this so I dunno.

Reply:
The thing is, the `(for/var/mob/murderer in range(10, src))` already does what you say. This is totally redundant.